src/dbnode/storage/flush.go

```diff
@@ -54,6 +54,20 @@ const (
 	flushManagerIndexFlushInProgress
 )
 
+type namespaceFlushes map[string]namespaceFlush
+
+type shardFlushes map[shardFlush]bool
+
+type namespaceFlush struct {
+	namespace    databaseNamespace
+	shardFlushes map[shardFlush]bool
```

Comment:
I would change this datastructure to:

1. Use only native Go types as the key and avoid using pointer types as part of the key for the map (the comparison of pointer types is not always as you'd expect especially when using interfaces, which we are here).
2. Use a map value to store the pointer types you're trying to access (since don't want to start it part of the key).

```go
// ...
  shardFlushes map[shardFlushKey]shardFlushValue
// ...

type shardFlushKey struct {
  shardID uint32
  blockStart xtime.UnixNano
}

type shardFlushValue struct {
  shard databaseShard
}
```

Reply:
Ah good idea - makes sense to key off the primitives. Since we only have one value I think we can use just this key struct for simplicity.